Oi! Eu vi o commit aqui no PR e acho que deu certo sim. Porém, em alguns testes, vi que tem outro erro que está acontecendo relacionado ao parâmetro `save_session=True` que buga quando você tenta colocar o mesmo e-mail para um ambiente de homologação e de produção, quando a sessão de um deles já está salva.

O traceback no caso é esse:

```
Traceback (most recent call last):
  File "c:\Users\Gustavo\Desktop\test.py", line 3, in <module>
    client = tabnews.Client('sowlfie@gmail.com', '***********')
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\Gustavo\Desktop\tabnews\__init__.py", line 59, in __init__
    self.enter_session(
  File "c:\Users\Gustavo\Desktop\tabnews\__init__.py", line 71, in enter_session
    self.dump_config(use_preview_tabnews_host)
  File "c:\Users\Gustavo\Desktopr\tabnews\mixins\auth.py", line 122, in dump_config
    if value[0][session['email']] and value[0][session['host']]:
```

Consegui resolver esse erro alterando essas linhas no `tabnews/mixins/auth.py`:

```python
value = check_EH(self.email, host)
for session in list_sessions:
    if value[0][session['email']] and value[1][session['host']]:
        user_not_saved = False
        session['session_id'] = session
```

para:
```python
value = check_EH(self.email, host)
for session in list_sessions:
    if value[0] == session['email'] and value[1] == session['host']:
        user_not_saved = False
        session['session_id'] = session
```

Daí, agradeceria se fizesse um commit também corrigindo isso, porque daí essa questão já é encerrada no PR. Desculpa por não ter avisado antes e desculpa por estar te dando trabalho nesse sentido.

Depois acho que eu preciso criar uma parte do repositório só para scripts de teste para testar tudo isso de uma vez, acho que seria o mais recomendado.

---

Sobre a atualização no PIP, eu geralmente faço via linha de comando depois que atualizo, mas acho que tem como configurar isso no GitHub para já ir automático.

Antes de enviar para o PIP, acho que precisaria atualizar a versão no `setup.py` no parâmetro `version="1.1.0"`, que acho que iria ser: `1.2.0`.

No mais, muito obrigado novamente! 🤝

--- a/tabnews/mixins/auth.py
+++ b/tabnews/mixins/auth.py
@@ -119,7 +119,7 @@ class LoginMixin:
 
             value = check_EH(self.email, host)
             for session in list_sessions:
-                if value[0][session['email']] and value[1][session['host']]:
+                if value[0] == session['email'] and value[1] == session['host']:
                     user_not_saved = False
                     session['session_id'] = session
 

--- a/tabnews/utils.py
+++ b/tabnews/utils.py
@@ -49,7 +49,6 @@ def get_preview_url():
         print('Não foi possível obter o host do homologação do Tabnews.')
         raise
 
-
 def url_validator(url):
     """
     Validate the URL.

